app/src/main/java/ch/epfl/sdp/Map/MapFragment.java

```diff
@@ -87,6 +88,11 @@ public void onServiceDisconnected(ComponentName name) {
             }
         };
 
+        // startService() overrides the default service lifetime that is managed by
+        // bindService(Intent, ServiceConnection, int):
+        // it requires the service to remain running until stopService(Intent) is called,
+        // regardless of whether any clients are connected to it.
+        ComponentName myService = getActivity().startService(new Intent(getContext(), LocationService.class));
```

Comment:
Isn't this a strange way of doing this ? Maybe not, I don't know. But with out tests failing because of contexts, we should try to avoid strange things.

Reply:
I found out that:
The android docs for startService state:
" Using startService() overrides the default service lifetime that is managed by bindService(Intent, ServiceConnection, int): it requires the service to remain running until stopService(Intent) is called, regardless of whether any clients are connected to it.
Whereas for bindService: the service will be considered required by the system only for as long as the calling context exists. For example, if this Context is an Activity that is stopped, the service will not be required to continue running until the Activity is resumed."

[source: https://stackoverflow.com/questions/1992676/activity-app-name-has-leaked-serviceconnection-serviceconnection-name438030]

Reply:
Ok and when do you call stopService ?

Reply:
Never for now, for the Location Service to always run in background. This was a quick fix of a bug:
"TabActivity has leaked ServiceConnection" when leaving the app on the emulator.
I didn't have the time to study this in details, it would be better to have the author of these lines check whether he didn't use the startService on purpose or not.

Reply:
Ah okay- anyway